gprofiler/python.py

```diff
@@ -110,6 +110,8 @@ def snapshot(self) -> ProcessToStackSampleCounters:
                     results[futures[future]] = future.result()
                 except StopEventSetException:
                     raise
+                except NoSuchProcess:
+                    logger.debug(f"Process process went down during profiling {futures[future]}", exc_info=True)
```

Comment:
Shouldn't it be a warning? What if the profiler crashed? We can't even check the output

Reply:
`NoSuchProcess` is most likely for interaction with the target process. It's much more likely that the target process exited (because processes exit, it happens).

Reply:
We don't use `psutil` on our profilers anyway

Reply:
> most likely

So it is possible that we could miss errors that we would like to log otherwise? Let's improve the reliability as much as possible if we can

Reply:
Note - I'm not missing any logs, just changed specific ones to `DEBUG` so they don't spam our backend with `ERROR`s that are, in 99.99% of the cases, just processes going down. In case there's a real problem with one of the profilers that causes processes to go down (e.g with async-profiler) - we will always have the full logs.

Reply:
Yeah I know, but I'm talking about that error that would happen in 0.01% of the time - maybe check for a certain REGEX on the output to make sure the error is okay?

Reply:
I don't know how will it look like when it's a real error. These logs will be able to help when debugging once we *know* there was a real problem.

They can also tell us there's a problem if it happens "every time" but that's more complex to catch. Therefore I think it's best kept at DEBUG